sklearn/model_selection/_split.py

```diff
@@ -913,6 +914,144 @@ def get_n_splits(self, X, y, groups):
         return int(comb(len(np.unique(groups)), self.n_groups, exact=True))
 
 
+class BaseRepeatedCrossValidator(with_metaclass(ABCMeta)):
+    """Base class for repeated cross validators
+
+    Implementations must define `_get_cv`.
+    """
+
+    def __init__(self, n_iter=2, n_splits=3):
+        self.n_iter = n_iter
+        self.n_splits = n_splits
+        self._random_states = []
+        for _ in range(n_splits):
+            random_state = check_random_state(
+                None).randint(np.iinfo(np.int32).max)
```

Comment:
It still needs to be possible to pass and int `random_state` for deterministic splitting.